modules/serverpod_auth/serverpod_auth_idp/serverpod_auth_idp_server/lib/src/providers/anonymous/business/utils/anonymous_idp_account_creation_util.dart

```diff
@@ -0,0 +1,130 @@
+import 'package:serverpod/serverpod.dart';
+
+import '../../../../../core.dart';
+import '../../../../../providers/email.dart' show RateLimit;
+import '../anonymous_idp_config.dart';
+
+/// {@template anonymous_idp_account_creation_util}
+/// Bundles utility function for the anonymous identity provider's account
+/// creation purposes.
+/// {@endtemplate}
+class AnonymousIdpAccountCreationUtil {
+  final AnonymousIdpAccountCreationUtilsConfig _config;
+  final DatabaseRateLimitedRequestAttemptUtil<String>? _rateLimitUtil;
+  final AuthUsers _authUsers;
+
+  /// Creates a new [AnonymousIdpAccountCreationUtil] instance.
+  AnonymousIdpAccountCreationUtil({
+    required final AnonymousIdpAccountCreationUtilsConfig config,
+    required final AuthUsers authUsers,
+  }) : _config = config,
+       _authUsers = authUsers,
+       _rateLimitUtil = config.rateLimit != null
+           ? DatabaseRateLimitedRequestAttemptUtil(
+               RateLimitedRequestAttemptConfig(
+                 domain: 'anonymous',
+                 source: 'account_creation',
+                 maxAttempts: config.rateLimit?.maxAttempts,
+                 timeframe: config.rateLimit?.timeframe,
+               ),
```

Comment:
Suggestion: Add the throw here to reduce the boilerplate below.

```suggestion
               RateLimitedRequestAttemptConfig(
                 domain: 'anonymous',
                 source: 'account_creation',
                 maxAttempts: config.rateLimit?.maxAttempts,
                 timeframe: config.rateLimit?.timeframe,
                 onRateLimitExceeded: (session, nonce) {
                   throw AnonymousAccountBlockedException(
                     reason: AnonymousAccountBlockedExceptionReason.throttled,
                   );
                 },
               ),
```